knative-operator/pkg/controller/knativeservingobsolete/knativeservingobsolete_controller.go

```diff
@@ -87,31 +92,106 @@ func (r *ReconcileKnativeServingObsolete) Reconcile(request reconcile.Request) (
 			return reconcile.Result{}, err
 		}
 	}
-	// Create the latest CR from the current (previous) CR
-	latest = &servingv1alpha1.KnativeServing{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      current.Name,
-			Namespace: current.Namespace,
-		},
-	}
-	latest.Spec.Config = current.Spec.Config
-	if err := common.Mutate(latest, r.client); err != nil {
+	// Orphan all the children by removing their OwnerRefs
+	if err := r.orphanObsoleteResources(current); err != nil {
 		return reconcile.Result{}, err
 	}
 	// Avoid a certs config conflict in the k-s controller
 	if err := r.removeOldCertsConfig(current.Namespace); err != nil {
 		return reconcile.Result{}, err
 	}
-	// Orphan the kids to avoid webhook race condition
-	if err := r.client.Delete(context.TODO(), current, client.PropagationPolicy(metav1.DeletePropagationOrphan)); err != nil {
-		return reconcile.Result{}, err
-	}
-	if err := r.client.Create(context.TODO(), latest); err != nil {
+
+	_, err := r.reconcileNewResource(current)
+	if err != nil {
 		return reconcile.Result{}, err
 	}
+
+	// if !equality.Semantic.DeepEqual(current.Status, new.Status) {
+	// 	current.Status.Version = new.Status.Version
+	// 	current.Status.Conditions = deepCopyConditions(new.Status.Conditions)
+	// 	if err := r.client.Status().Update(context.TODO(), current); err != nil {
+	// 		return reconcile.Result{}, err
+	// 	}
+	// 	return reconcile.Result{}, nil
+	// }
```

Comment:
This logic needs to stay intact I believe. It surfaces the status on the old thing to make it look like it used to before. Cloud Paks definitely depends on this.